### PR TITLE
Add port to default database setup.

### PR DESCRIPTION
**Pull request by grosch**
It's pretty common to need to change the port on the database.  Would be nice if the template handled that by default.

### CODE REVIEW

**Comment by grosch** [comment]
I don't really know how to test this, sorry :(

File: Sources/App/configure.swift
Comment (mattpolzin) [inline Sources/App/configure.swift:11]: this gets defined under postgres configuration but used under both postgres and mysql configuration below so that's going to be a problem. Maybe this code should live under all database configurations (there's a way in mustache to ask whether the template is using any database at all, but I forget what it is off the top of my head) and it should set an optional `port` property that can be nil coalesced below to the appropriate port for each given database (`5432` for postgres and `3306` for mysql).

**Review by mattpolzin — CHANGES_REQUESTED** [review]

File: Sources/App/configure.swift
Comment (mattpolzin) [reply Sources/App/configure.swift:11]: Found it; the logic used by the `docker-compose.yml` file is "configured for Fluent but not configured for SQLite."

```
{{#fluent}}{{^db.is_sqlite}}
  ...
{{/db.is_sqlite}}{{/fluent}}
```

File: Sources/App/configure.swift
Comment (0xTim) [inline Sources/App/configure.swift:31]: This will break as there's no port variable defined if we want to use MySQL

**Comment by 0xTim** [comment]
I'm not 100% convinced on the need for this. The template shows how to provide configuration for the database and those that need to change it can. I think adding `port: 5432` may be useful however. Will open it up to the community for comment, happy to be persuaded

File: Sources/App/configure.swift
Comment (gwynne) [inline Sources/App/configure.swift:31]: I suggest doing it this way instead:
```suggestion
    {{#fluent.db.is_postgres}}app.databases.use(.postgres(
        hostname: Environment.get("DATABASE_HOST") ?? "localhost",
        port: Environment.get("DATABASE_PORT).flatMap(Int.init(_:)) ?? 5432, // TODO: Add `PostgresConfiguration.ianaPortNumber` like MySQL has
        username: Environment.get("DATABASE_USERNAME") ?? "vapor_username",
        password: Environment.get("DATABASE_PASSWORD") ?? "vapor_password",
        database: Environment.get("DATABASE_NAME") ?? "vapor_database"
    ), as: .psql){{/fluent.db.is_postgres}}{{#fluent.db.is_mysql}}app.databases.use(.mysql(
        hostname: Environment.get("DATABASE_HOST") ?? "localhost",
        port: Environment.get("DATABASE_PORT").flatMap(Int.init(_:)) ?? MySQLConfiguration.ianaPortNumber,
```
Reasons:

- This correctly handles both PostgreSQL and MySQL.
- If the specified port is not a valid integer, this will use the default instead of crashing. This is what happens if any of the other env vars values' are invalid - we don't validate that the host is a valid DNS name or that the database name is a valid Postgres or MySQL identifier, so silently ignoring the port if it's invalid is consistent. It's also much simpler.
- No additional database-specific stanzas are needed.
- With MySQL, the fallback value is not a hardcoded number; the only reason it is for Postgres is I haven't added the exported constant to that driver yet.

**Review by gwynne — CHANGES_REQUESTED** [review]
I think this is worth having, but should be done a little differently - see my inline suggestion for details.

File: Sources/App/configure.swift
Comment (0xTim) [reply Sources/App/configure.swift:31]: I think this is a good compromise and cleaner, happy to go with this solution

**Review by mattpolzin — COMMENTED** [review]
Naturally, I defer to the Core Team (so I've changed my review to a comment). I think Gwynne's solution is very clean.

File: Sources/App/configure.swift
Comment (grosch) [reply Sources/App/configure.swift]: I went with Gwynne's suggestion.

File: Sources/App/configure.swift
Comment (grosch) [reply Sources/App/configure.swift]: Went with Gwynne's suggestion.

File: Sources/App/configure.swift
Comment (gwynne) [inline Sources/App/configure.swift:10]: Sorry, looks like this extra line snuck into one of my suggestion comments, once it's gone this is ready for merge.
```suggestion
```

File: Sources/App/configure.swift
Comment (0xTim) [inline Sources/App/configure.swift:13]: This can now be updated to use `PostgresConfiguration.ianaPortNumber`

**Review by 0xTim — CHANGES_REQUESTED** [review]

File: Sources/App/configure.swift
Comment (gwynne) [inline Sources/App/configure.swift:13]:
```suggestion
        port: Environment.get("DATABASE_PORT").flatMap(Int.init(_:)) ?? PostgresConfiguration.ianaPortNumber,
```

**Review by 0xTim — APPROVED** [review]
LGTM

**Comment by grosch** [comment]
This is blocked because of Gwynne and Matt's changes, but they were incorporated.  Can somebody provide guidance on how I close those out so this can merge please?

**Comment by mattpolzin** [comment]
@grosch I don't follow -- if you are talking about me, what changes of mine are blocking this? My suggestions were superseded by Gwynne's suggestions, if you are referring to the changes I originally suggested.

**Comment by grosch** [comment]
@mattpolzin At the bottom of the comments thread it shows in red that changes are requested.  If I expand that I see you and @gwynne listed.  Then it says Merging is blocked and that it can be performed once requested changes are addressed.  

I don't see how to resolve that as the items both of you put were resolved.

**Review by mattpolzin — APPROVED** [review]

**Comment by mattpolzin** [comment]
Huh, this appears to be a new GitHub thing. In the past, sitting in a unreviewed state has been enough -- now I must explicitly approve the PR after having previously requested changes. 🤷

**Comment by grosch** [comment]
Thanks, Matt.  @gwynne Anything else needed from you?  If no, would you re-approve please?

**Review by gwynne — APPROVED** [review]

**Pull request merged**
